src/colvaratoms.h

```diff
@@ -257,8 +257,25 @@ class colvarmodule::atom_group
   /// \brief Index in the colvarproxy arrays (if the group is scalable)
   int index;
 
+  /// \brief The forces acting on the group atoms (stored mainly used for calculating the fitting group forces)
+  std::vector<cvm::rvector> group_forces;
+
 public:
 
+  class group_force_object {
+  public:
+    group_force_object(cvm::atom_group* ag);
+    ~group_force_object();
+    void add_atom_force(size_t i, const cvm::rvector& force);
+  private:
+    cvm::atom_group* m_ag;
+    cvm::atom_group* m_group_for_fit;
+    bool m_has_fitting_force;
+    void apply_force_with_fitting_group();
+  };
+
+  group_force_object get_group_force_object();
+
```

Comment:
My intention is to use the RAII mechanism to call `apply_force_with_fitting_group()` automatically when the object is out of scope so that I don't need to call something like `add_atom_force_done()`, and also try my best to minimize the change to `atom_group`. I believe the creation of the object should be cheap enough (three pointer fields and a boolean field). If you don't like it I can fuse it into `cvm::atom_group`.